### PR TITLE
Consistently format the order notes for refunds and charges

### LINKED ISSUE #8345: The amount format in the refunded order notes does not honor currency format options for refunds triggered on the order details page
### Describe the bug

There are two ways to issue a refund for an order:

- Via WP Admin, using the Refund button in the order details page
- Externally, using the Stripe Dashboard

In this PR, https://github.com/Automattic/woocommerce-payments/pull/8200 we implemented the support to handle external refunds (in other words, we implemented a webhook listener for refund events). After that, we noticed that the refunded amount displayed in the refunded order notes was different depending on how the refund was initiated, but only with the following currency options enabled:
 - Currency position: Right
 - Number od decimals: 0

### To Reproduce

1. Go to WooCommerce > Settings and set `Currency position: Right` and `Number of decimals: 0`
2. As a shopper, place an order.
3. As a merchant, issue a refund for the same order

### Actual behavior

The refunded amount in the order notes does not follow the currency formatting previously selected (see screenshot below)

### Screenshots

310346195-53803bde-c1c1-4522-a7db-fd65ece6a6db

### Additional context

- Refunds triggered from WP Admin are handled in WC_Payment_Gateway_WCPay::process_refund
- It's also worth checking the refund_line_items method https://github.com/Automattic/woocommerce.com/blob/c1833bf2c343be095c7fd6975a26c6cf8a57967b/plugins/woocommerce/includes/class-wc-ajax.php#L2137

### PR DESCRIPTION
**Pull request by deepakpathania**
Fixes #8345 

#### Changes proposed in this Pull Request

- Add a helper method for formatting the order notes consistently based on whether MC is enabled or not.
- Update the usage in order service for `charge` and `refund` notes.

#### Testing instructions

* Follow the steps mentioned in #8345 and ensure the behavior is consistent.
* Also disable MC and ensure the store settings are used, whereas with MC enabled, MC settings override the store ones.

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10036` or branch name `feature/consistent-note-formatting` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=feature/consistent-note-formatting) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8b2da266fb50b3ec3e2549e3b14f8bae6a5bc87d
- Build time: 2024-12-24 11:51:34 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-order-service.php [hunk 1]
```diff
@@ -1930,7 +1928,14 @@ private function complete_order_processing( $order, $intent_status = null ) {
 	 * @return string The formatted order total.
 	 */
 	private function get_order_amount( $order ) {
-		return WC_Payments_Explicit_Price_Formatter::get_explicit_price( wc_price( $order->get_total(), [ 'currency' => $order->get_currency() ] ), $order );
+		$multi_currency_instance = WC_Payments_Multi_Currency();
+		$order_price             = $order->get_total();
+		if ( ! $order_price ) {
+			return $order_price;
+		}
```
Comment (tpaksu) [inline includes/class-wc-payments-order-service.php]: To match the previous state of the code, we can remove this check, as the order total will always be passed through `wc_price`.

File: includes/multi-currency/MultiCurrency.php [hunk 2]
```diff
@@ -1326,6 +1326,31 @@ public function is_initialized(): bool {
...
+	/**
+	 * Returns the amount with the backend format.
+	 *
+	 * @param float $amount The amount to format.
+	 * @param array $args The arguments to pass to wc_price.
+	 *
+	 * @return string The formatted amount.
+	 */
+	public function get_backend_formatted_wc_price( float $amount, array $args = [] ): string {
+		if ( ! self::has_additional_currencies_enabled() ) {
```
Comment (tpaksu) [inline includes/multi-currency/MultiCurrency.php]: To be safe, I think we also need to check `WC_Payments_Features::is_customer_multi_currency_enabled()`.

**Review by tpaksu — COMMENTED** [review]
Thanks for working on this, I've found two issues in the code, which may require more changes later. I'll test it manually after we fix them all.

File: includes/class-wc-payments-order-service.php (hunk 1 above)
Comment (deepakpathania) [reply includes/class-wc-payments-order-service.php]: Fair, removed.

File: includes/multi-currency/MultiCurrency.php (hunk 2 above)
Comment (deepakpathania) [reply includes/multi-currency/MultiCurrency.php]: Added a check for both.

**Review by tpaksu — APPROVED** [review]
Tested with MC enabled, refund from Stripe: 

image

Tested with MC Enabled refund via Order details screen:

image

Tested with MC disabled via removing the additional currencies

image

The currencies have consistent formatting, matching the rest of the UI now. But beware that the order notes are static, and won't change once it's created, and if we enable/disable multi-currency after a note is created, it won't change its formatting. 

The changes look good to me, thanks for working on this issue. We can 🚢  it.

**Pull request merged**